README.md

```diff
@@ -1,46 +1,114 @@
-# Model conversion of Kim Model
-### PyTorch -> ONNX -> MxNet (Python) -> MxNet (Scala)
-
+# Kim Model in ONNX
+##### PyTorch -> ONNX -> MXNet (Python) -> MXNet (Scala) -> Spark Distribued Inference (Scala)
 ----------
-### Build the docker image and run a container
+### Docker Image Build
 ```
 docker build --tag h379wang/kim-model-conversion .
-docker run -it h379wang/kim-model-conversion:latest /bin/bash
+```
+### Run Docker Container
+Assume that from `ONNX-demo`, [Castor-data](https://git.uwaterloo.ca/jimmylin/Castor-data) is checked out in `..` . Assume the current folder is `ONNX-demo`.
+```
+docker run -it -v $(pwd)/../Castor-data:/data h379wang/kim-model-conversion:latest /bin/bash
+```
+### Model Conversion
+PyTorch -> ONNX -> MXNet Python -> MXNet Scala
+```
+source /opt/intel/bin/compilervars.sh intel64
+python3 export_models.py
+# Move the generated metadata of scala version of Kim Model
+mv *.params scala_inference/
+mv *.json scala_inference/
 ```
 ----------
-### Get Datasets
+### Inference Under Embedding Subset for SST-1 Dataset
+##### PyTorch Inference
+```
+python3 pytorch_inference.py --embedding /data/embeddings/word2vec/word2vec.sst-1 --seed 1
 ```
-cd kim_cnn
-chmod +x get_data.sh
-./get_data.sh
+##### MXNet Python Inference
 ```
+source /opt/intel/bin/compilervars.sh intel64
+python3 mxnet_inference.py --embedding /data/embeddings/word2vec/word2vec.sst-1 --seed 1
+```
+#####  MXNet Scala Inference
+```
+cd /app/scala_inference/
+export LD_LIBRARY_PATH="/app/incubator-mxnet/scala-package/native/linux-x86_64-cpu/target:$LD_LIBRARY_PATH"
+export MXNET_JAR_PATH="/app/incubator-mxnet/scala-package/core/target/*.jar,/app/incubator-mxnet/scala-package/spark/target/*.jar"
+export PATH="/app/spark-2.3.1-bin-hadoop2.7/bin:$PATH"
+mvn package
 
-Note that the convert_bin2txt.py process of converting the whole GoogleNews-vectors-negative300.bin embedding can take a long time. For testing purpose if you only need a small subset of the embedding, please execute the following commands:
+spark-submit --conf spark.driver.extraLibraryPath="${LD_LIBRARY_PATH}" \
```

Comment:
Spark is used to run normal Scala inference now?

Reply:
Yes just for convenience.